apps/src/code-studio/progressRedux.js

```diff
@@ -112,6 +116,26 @@ export default function reducer(state = initialState, action) {
     };
   }
 
+  if (action.type === SET_CURRENT_LEVEL_ID) {
+    return {
+      ...state,
+      currentLevelId: action.levelId
+    };
+  }
+
+  if (action.type === NAVIGATE_TO_LEVEL_ID) {
+    const newLevel = state.lessons[0].levels.find(level =>
+      level.ids.find(id => id === action.levelId)
+    );
+
+    handleNavigateToLevel(state, newLevel.url, action.levelId);
```

Comment:
we should use a [thunk](https://redux.js.org/usage/side-effects-approaches#thunks) (or something similar) here instead of having a [side effect](https://redux.js.org/usage/side-effects-approaches) in a reducer

Reply:
As we fully integrate with projects, that thunk could also be where we fetch the channel id for the new level.

Reply:
looks good to me!